homeassistant/components/climate/xiaomi_miio.py

```diff
@@ -0,0 +1,624 @@
+            "dehum": "3",
+            "airSup": "4"
+        },
+        "wi": {
+            "auto": "3",
+            "1": "0",
+            "2": "1",
+            "3": "2"
+        },
+        "sw": {
+            "on": "0",
+            "off": "1"
+        },
+        "tt": "1",
+        "li": {
+            "off": "a0"
+        }
+    },
+    "0180111111": {
+        "des": "media_1",
+        "main": "0180111111pomowiswtt02"
+    },
+    "0180222221": {
+        "des": "gree_1",
+        "main": "0180222221pomowiswtt02"
+    },
+    "0100010727": {
+        "des": "gree_2",
+        "main": "0100010727pomowiswtt1100190t0t20500\
+                2102000t6t0190t0t207002000000t4wt0",
+        "off": "010001072701011101004000205002112000\
+                D04000207002000000A0",
+        "EXTRA_VALUE": ["t0t", "t6t", "t4wt"],
+        "t0t": "1",
+        "t6t": "7",
+        "t4wt": "4"
+    },
+    "0100004795": {
+        "des": "gree_8",
+        "main": "0100004795pomowiswtt0100090900005002"
+    },
+    "0180333331": {
+        "des": "haier_1",
+        "main": "0180333331pomowiswtt12"
+    },
+    "0180666661": {
+        "des": "aux_1",
+        "main": "0180666661pomowiswtt12"
+    },
+    "0180777771": {
+        "des": "chigo_1",
+        "main": "0180777771pomowiswtt12"
+    }
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+    vol.Required(CONF_SENSOR, default=None): cv.entity_id,
+    vol.Optional(CONF_CUSTOMIZE, default=None): dict,
+    vol.Optional(CONF_SYNC, default=15): cv.positive_int
+})
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Set up the smart mi acpartner platform."""
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME) or DEFAULT_NAME
+    token = config.get(CONF_TOKEN)
+    sensor_entity_id = config.get(CONF_SENSOR)
+    # target_temp = config.get(CONF_TARGET_TEMP)
+    sync = config.get(CONF_SYNC)
+    customize = config.get(CONF_CUSTOMIZE)
+
+    add_devices_callback([MiAcPartner(
+        hass, name, None, None, None, 'auto', None,
+        'off', 'off', None, DEFAULT_MAX_TMEP, DEFAULT_MIN_TMEP, host,
+        token, sensor_entity_id, sync, customize)])
+
+
+class ClimateStatus(ClimateDevice):
+    """Container for status reports from the climate."""
+    def __init__(self, data):
+        self.data = data
+
+    @property
+    def acpower(self):
+        """Get acpower from the climate."""
+        return self.data[2]
+
+    @property
+    def acmodel(self):
+        """Get acmodel from the climate."""
+        return str(self.data[0][0:2] + self.data[0][8:16])
+
+    @property
+    def power(self):
+        """Get power from the climate."""
+        return self.data[1][2:3]
+
+    @property
+    def mode(self):
+        """Get mode from the climate."""
+        return self.data[1][3:4]
+
+    @property
+    def wind_force(self):
```

Comment:
What is a "wind force"? The strength of a fan? Shouldn't this be `fan_mode` then?